x/lightclient/types/params.go

```diff
@@ -2,78 +2,84 @@ package types
 
 import (
 	"bytes"
+	"errors"
+	"fmt"
 	"time"
 
 	"github.com/cometbft/cometbft/libs/math"
 	ibcclienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
 	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
 	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
 	ics23 "github.com/cosmos/ics23/go"
+	sequencertypes "github.com/dymensionxyz/dymension/v3/x/sequencer/types"
 )
 
+func DefaultExpectedCanonicalClientParams() ibctm.ClientState {
+	return ExpectedCanonicalClientParams(sequencertypes.DefaultUnbondingTime)
+}
+
 // ExpectedCanonicalClientParams defines the expected parameters for a canonical IBC Tendermint client state
 // The ChainID is not included as that varies for each rollapp
 // The LatestHeight is not included as there is no condition on when a client can be registered as canonical
-var ExpectedCanonicalClientParams = ibctm.ClientState{
-	// Trust level is the fraction of the trusted validator set
-	// that must sign over a new untrusted header before it is accepted.
-	// For a rollapp should be 1/1.
-	TrustLevel: ibctm.NewFractionFromTm(math.Fraction{Numerator: 1, Denominator: 1}),
-	// TrustingPeriod is the duration of the period since the
-	// LatestTimestamp during which the submitted headers are valid for update.
-	TrustingPeriod: time.Hour * 24 * 7 * 2,
-	// Unbonding period is the duration of the sequencer unbonding period.
-	UnbondingPeriod: time.Hour * 24 * 7 * 3,
-	// MaxClockDrift defines how much new (untrusted) header's Time
-	// can drift into the future relative to our local clock.
-	MaxClockDrift: time.Minute * 10,
-	// Frozen Height should be zero (default) as frozen clients cannot be canonical
-	// as they cannot receive state updates
-	FrozenHeight: ibcclienttypes.ZeroHeight(),
-	// ProofSpecs defines the ICS-23 standard proof specifications used by
-	// the light client. It is used configure a proof for either existence
-	// or non-existence of a key value pair
-	ProofSpecs:                   commitmenttypes.GetSDKSpecs(),
-	AllowUpdateAfterExpiry:       false,
-	AllowUpdateAfterMisbehaviour: false,
-	// For chains using Cosmos-SDK's default x/upgrade module, the upgrade path is as follows
-	UpgradePath: []string{"upgrade", "upgradedIBCState"},
+// AllowUpdateAfterExpiry and AllowUpdateAfterMisbehaviour are not checked, they are deprecated
+func ExpectedCanonicalClientParams(rollappUnbondingPeriod time.Duration) ibctm.ClientState {
+	return ibctm.ClientState{
+		// Trust level is the fraction of the trusted validator set
+		// that must sign over a new untrusted header before it is accepted.
+		// At LEAST this much must sign over the untrusted header. Voting sets all have power
+		// 1, so at least 1/3 of power 1 is 1.
+		TrustLevel: ibctm.NewFractionFromTm(math.Fraction{Numerator: 1, Denominator: 3}),
```

Comment:
I know 1/3 is still valid, but don't you think 1/1 is cleaner? what's the reason to change it back to 1/3?

Reply:
Tendermint bug
Doesn't support 1/1, because they do a > operation (not >=)
I'm submitting to hackerone